OpenRA.Mods.Common/Traits/Render/DrawLineToTarget.cs

```diff
@@ -62,65 +43,61 @@ void INotifySelected.Selected(Actor a)
 			lifetime = info.Delay;
 		}
 
+		void INotifySelected.Selected(Actor a)
+		{
+			ShowTargetLines(a);
+		}
+
 		IEnumerable<IRenderable> IRenderAboveShroudWhenSelected.RenderAboveShroud(Actor self, WorldRenderer wr)
 		{
-			var force = Game.GetModifierKeys().HasModifier(Modifiers.Alt);
-			if ((lifetime <= 0 || --lifetime <= 0) && !force)
+			if (self.Owner != self.World.LocalPlayer)
 				yield break;
 
-			if (!(force || Game.Settings.Game.DrawTargetLine))
+			// shift is "force" too, players want to see the lines when in waypoint mode.
+			bool force = Game.GetModifierKeys().HasModifier(Modifiers.Alt)
+			          || Game.GetModifierKeys().HasModifier(Modifiers.Shift);
+
+			if ((lifetime <= 0 || --lifetime <= 0) && !force)
```

Comment:
Can't this be simplified to `if (--lifetime <= 0 && !force)`?

Reply:
Fixed.